packages/loader/container-loader/src/container.ts

```diff
@@ -1251,17 +1254,19 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     }
 
     private get client(): IClient {
-        const client: IClient = this.options?.client !== undefined
-            ? (this.options.client as IClient)
-            : {
-                details: {
-                    capabilities: { interactive: true },
-                },
-                mode: "read", // default reconnection mode on lost connection / connection error
-                permission: [],
-                scopes: [],
-                user: { id: "" },
-            };
+        const client: IClient & {containerSessionId: string} =
```

Comment:
>& {containerSessionId: string} [](start = 30, length = 30)

Similar to Vlad's earlier comment about it not being clear how this is used, it's a bit of an adventure tracing how this getter gets used and consumed all the way into the document delta connection.  It feels like that path is being abused to shoehorn this information into the messages, and that if we want to use it this way we should add it into IClient (though idk if the container session is inherently coupled with the client).  Though I totally understand why it's being done this way so feel free to ignore 👀

Reply:
added some comments